vnext/Microsoft.ReactNative/CoreApp.cpp

```diff
@@ -0,0 +1,306 @@
+        }
+      }
+
+      if (auto res = xaml::ResourceDictionary(rnca->resourcesAbi, winrt::take_ownership_from_abi)) {
+        app.Resources(res);
+      } else {
+        try {
+          auto xcr = winrt::Microsoft::UI::Xaml::Controls::XamlControlsResources();
+          app.Resources(xcr);
+        } catch (...) {
+        }
+      }
+    });
+    impl_app->ViewCreatedInternal([](react::ReactApplication const &app, winrt::hstring const &args) {
+      auto rootFrame = xaml::Window::Current().Content().as<xaml::Controls::Frame>();
+      rootFrame.Navigate(winrt::xaml_typename<react::CoreAppPage>(), winrt::box_value(args));
+    });
+    impl_app->PageNavigatedInternal([rnca](react::ReactApplication const & /*app*/, react::ReactRootView const &view) {
+      view.ComponentName(rnca->componentName);
+    });
+  });
+}
+
+void SetFromJson(JsonObject json, wchar_t const *key, wchar_t const *&appValue) {
+  if (auto v = json.TryLookup(key); v && v.ValueType() == JsonValueType::String) {
+    appValue = _wcsdup(v.GetString().c_str());
+  }
+}
+
+void SetFromJson(JsonObject json, wchar_t const *key, bool &appValue) {
+  if (auto v = json.TryLookup(key); v && v.ValueType() == JsonValueType::Boolean) {
+    appValue = v.GetBoolean();
+  }
+}
+
+template <typename T>
+std::enable_if_t<std::is_arithmetic_v<T>> SetFromJson(JsonObject json, wchar_t const *key, T &appValue) {
+  if (auto v = json.TryLookup(key); v && v.ValueType() == JsonValueType::Number) {
+    appValue = static_cast<T>(v.GetNumber());
+  }
+}
+
+#define SET_FROM_JSON(key) SetFromJson(json, L#key, app->key)
+
+JsonObject LoadJsonFromFile(wchar_t const *jsoncPath) {
+  std::wifstream appConfigJson(jsoncPath);
+  winrt::check_bool(appConfigJson.good());
+  std::wostringstream out;
+  out << appConfigJson.rdbuf();
+  auto input = out.str();
+
+  // strip comments from jsonc since WinRT's JsonObject doesn't support them
+  // see
+  // https://github.com/microsoft/vscode/blob/432349e1eb4a90eccd1039c61f719e6f54c77607/src/vs/base/common/stripComments.js
+  auto comments = std::wregex(
+      LR"(("[^"\\]*(?:\\.[^"\\]*)*")|('[^'\\]*(?:\\.[^'\\]*)*')|(\/\*[^\/\*]*(?:(?:\*|\/)[^\/\*]*)*?\*\/)|(\/{2,}.*?(?:(?:\r?\n)|$)))");
+  std::wstring stripped;
+  std::size_t previous = 0;
+  for (auto it = std::wsregex_iterator(input.begin(), input.end(), comments); it != std::wsregex_iterator(); ++it) {
+    auto match = *it;
+    stripped += input.substr(previous, match.position() - previous);
+    if (match[3].matched) {
+      // do nothing
+    } else if (match[4].matched) {
+      if (*(match[4].second - 1) == L'\n') {
+        stripped += *(match[4].second - 2) == L'\r' ? L"\r\n" : L"\n";
+      }
+    } else {
+      stripped += match.str();
+    }
+
+    previous = match.position() + match.length();
+  }
+
+  stripped += input.substr(previous);
+
+  auto json = JsonObject::Parse(stripped);
+  return json;
+}
+
+static void SetPackageProviders(const JsonObject &json, RNCoreApp *app) {
```

Comment:
>const [](http://example.com/codeflow?start=32&length=5)

nit: are we left or right const? It is better to be consistent in the same code file.